src/metabase/lib/drill_thru/underlying_records.cljc

```diff
@@ -45,6 +53,42 @@
    :table-name table-name})
 
 (defmethod lib.drill-thru.common/drill-thru-method :drill-thru/underlying-records
-  [_query _stage-number _drill-thru & _]
-  (throw (ex-info "Not implemented" {}))
-  #_(lib.filter/filter query stage-number (lib.options/ensure-uuid [(keyword filter-op) {} (lib.ref/ref column) value])))
+  [query stage-number {:keys [column-ref dimensions]} & _]
+  (let [top-query   (lib.underlying/top-level-query query)
+        ;; Drop all aggregations, breakouts, sort orders, etc. to get the underlying records.
+        base-query  (lib.util/update-query-stage top-query stage-number
+                                                 dissoc :aggregation :breakout :order-by :limit :fields)
```

Comment:
Do we trust that `top-query` has a stage at `stage-number`?

Reply:
No, I think this is a bug. It should be working with the last stage of the top-level query. I've updated this.